src/DoocsIfff.cc

```diff
@@ -30,39 +26,37 @@ namespace ChimeraTK {
       boost::shared_ptr<NDRegisterAccessor<float>> const& f1Value,
       boost::shared_ptr<NDRegisterAccessor<float>> const& f2Value,
       boost::shared_ptr<NDRegisterAccessor<float>> const& f3Value, DoocsUpdater& updater)
-  : D_ifff(doocsPropertyName, eqFct), _i1Value(i1Value), _f1Value(f1Value), _f2Value(f2Value), _f3Value(f3Value),
-    _updater(updater), _eqFct(eqFct), isWriteable(_i1Value->isWriteable() && _f1Value->isWriteable() &&
-                                          _f2Value->isWriteable() && _f3Value->isWriteable()) {
+  : D_ifff(doocsPropertyName, eqFct), PropertyBase(eqFct, doocsPropertyName, updater), _i1Value(i1Value),
+    _f1Value(f1Value), _f2Value(f2Value), _f3Value(f3Value) {
     checkSourceConsistency();
     registerIfffSources();
-    if(_i1Value->isReadOnly() && _f1Value->isReadOnly() && _f2Value->isReadOnly() && _f3Value->isReadOnly()) {
-      this->set_ro_access();
-    }
   }
 
   void DoocsIfff::checkSourceConsistency() {
     bool areAllSourcesWritable =
         (_i1Value->isWriteable() && _f1Value->isWriteable() && _f2Value->isWriteable() && _f3Value->isWriteable());
+    _isWriteable = areAllSourcesWritable;
     bool areAllSourcesReadOnly =
         (_i1Value->isReadOnly() && _f1Value->isReadOnly() && _f2Value->isReadOnly() && _f3Value->isReadOnly());
 
     if(!areAllSourcesWritable && !areAllSourcesReadOnly) {
       ChimeraTK::logic_error("Doocs Adapter IFFF configuration Error: some IFFF sources are not writable");
     }
+    if(areAllSourcesReadOnly) {
+      this->set_ro_access();
+    }
   }
 
   void DoocsIfff::registerIfffSources() {
     registerVariable(OneDRegisterAccessor<int>(_i1Value));
     registerVariable(OneDRegisterAccessor<float>(_f1Value));
     registerVariable(OneDRegisterAccessor<float>(_f2Value));
     registerVariable(OneDRegisterAccessor<float>(_f3Value));
+    _mainOutputVar = _i1Value;
```

Comment:
In principle, all 4 components of the IFFF are equal. In which way does this introduce a visible asymmetry? If it doesn't, consider renaming "_mainOutputVar".

Reply:
renamed _mainOutputVar to clarify